react-components/src/hooks/useSearchAssetsMapped360Annotations.tsx

```diff
@@ -0,0 +1,146 @@
+    async () => {
+      if (query === '') {
+        const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+        return assetMappings;
+      }
+      const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+
+      const filteredSearchedAssets =
+        assetMappings.filter((asset) => {
+          const isInName = asset.name.toLowerCase().includes(query.toLowerCase());
+          const isInDescription = asset.description?.toLowerCase().includes(query.toLowerCase());
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      return filteredSearchedAssets;
+    },
+    {
+      staleTime: Infinity
+    }
+  );
+};
+
+async function getAssetsMapped360Annotations(
+  sdk: CogniteClient,
+  siteIds: string[]
+): Promise<Asset[]> {
+  const fileIdsList = await get360ImagesFileIds(siteIds, sdk);
+  const image360Annotations = await get360ImageAnnotations(fileIdsList, sdk);
+  const result = await get360AnnotationAssets(image360Annotations, sdk);
+
+  return result;
+}
+
+async function get360AnnotationAssets(
+  image360Annotations: AnnotationModel[],
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const annotationMapping = image360Annotations.map((annotation) => ({
+    assetId:
+      ((annotation.data as AnnotationsBoundingVolume).assetRef?.id as number) ??
+      ((annotation.data as AnnotationsBoundingVolume).assetRef?.externalId as string)
+  }));
+
+  const uniqueAnnotationMapping = uniqBy(annotationMapping, 'assetId');
+
+  const assets = chunk(uniqueAnnotationMapping, 1000).map(async (uniqueAssetsChunk) => {
+    const retrievedAssets = await sdk.assets.retrieve(
+      uniqueAssetsChunk.map(({ assetId }) => ({
+        [typeof assetId === 'number' ? 'id' : 'externalId']: assetId
+      })) as unknown as IdEither[],
+      { ignoreUnknownIds: true }
+    );
+    return retrievedAssets;
+  });
+
+  return (await Promise.all(assets)).flat();
+}
+
+async function get360ImagesFileIds(siteIds: string[], sdk: CogniteClient): Promise<number[]> {
+  const fileIdListPromises = siteIds.map(async (siteId) => {
+    const req = {
+      metadata: { site_id: siteId }
+    };
+    const fileIds = await listFileIds(req as FileFilterProps, sdk);
+    return fileIds;
+  });
+
+  const fileIdsList = (await Promise.all(fileIdListPromises)).flat();
+
+  return fileIdsList;
+}
+
+async function get360ImageAnnotations(
+  fileIdsList: number[],
+  sdk: CogniteClient
+): Promise<AnnotationModel[]> {
+  const annotationArray = await Promise.all(
+    chunk(fileIdsList, 1000).map(async (fileIdsChunk) => {
+      const filter: AnnotationFilterProps = {
+        annotatedResourceIds: fileIdsChunk?.map((id) => ({ id })) ?? [],
+        annotatedResourceType: 'file',
+        annotationType: 'images.AssetLink'
+      };
+      const annotations = await sdk.annotations
+        .list({
+          limit: 1000,
+          filter
+        })
+        .autoPagingToArray({ limit: Infinity });
```

Comment:
Why do you need autoPaging here if you always fetch <=1000 annotations, so one page exactly? Because autoPaging will fetch all annotations, as far as I can tell, you are fetching all annotations for each chunk of initial array